### PR TITLE
HGCal EGamma ID helpers and Phase-II MiniAOD 93x

### PR DESCRIPTION
**Pull request by nsmith-**
backport of #21037

### CODE REVIEW

**Comment by kpedro88** [comment]
assign upgrade

**Comment by kpedro88** [comment]
please test

File: PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py [hunk 1]
```diff
@@ -27,6 +27,13 @@ def miniAOD_customizeCommon(process):
 
     process.patMuons.computeMiniIso = cms.bool(True)
     
+    # EGamma objects from HGCal are not yet in GED
+    # so add companion collections for Phase-II MiniAOD production
+    from Configuration.Eras.Modifier_phase2_hgcal_cff import phase2_hgcal
+    from PhysicsTools.PatAlgos.slimming.addHGCalEgammaCollections import addHGCalEgammaCollections
+    if phase2_hgcal.isChosen():
```
Comment (slava77) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: @kpedro88 
I recall that you mentioned (in ORP?) that 93X updates to miniAOD will come without modifying the default miniAOD of 93X.
This PR changes the default.
Please clarify.

Comment (kpedro88) [reply PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: @slava77 though the use of eras can be improved (as commented in the master branch PR), this doesn't change the default 93X miniAOD.

**Comment by slava77** [comment]
On 10/31/17 5:54 PM, Kevin Pedro wrote:
> *@kpedro88* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py
> <https://github.com/cms-sw/cmssw/pull/21045#discussion_r148020927>:
> 
>> @@ -27,6 +27,13 @@ def miniAOD_customizeCommon(process):
>  
>      process.patMuons.computeMiniIso = cms.bool(True)
>      
> +    # EGamma objects from HGCal are not yet in GED
> +    # so add companion collections for Phase-II MiniAOD production
> +    from Configuration.Eras.Modifier_phase2_hgcal_cff import phase2_hgcal
> +    from PhysicsTools.PatAlgos.slimming.addHGCalEgammaCollections import addHGCalEgammaCollections
> +    if phase2_hgcal.isChosen():
> 
> @slava77 <https://github.com/slava77> though the use of eras can be
> improved (as commented in the master branch PR), this doesn't change the
> default 93X miniAOD.


are you saying that phase2_hgcal is not used in the standard upgrade
workflows for miniAOD?

**Comment by kpedro88** [comment]
@slava77 I guess I'm confused about your comment. I thought by "default miniAOD" you meant Run2. Of course this will change the upgrade miniAOD for 93X, that is the point of the PR.

My comment in the ORP only covered MIP timing-related changes.

**Comment by slava77** [comment]
On 10/31/17 6:35 PM, Kevin Pedro wrote:
> @slava77 <https://github.com/slava77> I guess I'm confused about your
> comment. I thought by "default miniAOD" you meant Run2. Of course this
> will change the upgrade miniAOD for 93X, that is the point of the PR.
> 
> My comment in the ORP only covered MIP timing-related changes.

"default" for me is everything in the current release covered by
runTheMatrix

OK, so the answer from you is that we can expect changes in the upgrade wfs
for 93X.

**Comment by kpedro88** [comment]
@nsmith- please bring this up to date with #21037

**Comment by slava77** [comment]
@nsmith- 
please update this PR to by in sync with #21037

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/21045	91767d3
- backport for miniAOD production for HGCAL "FromMultiCl". Changes in the code are the same as in the master branch PR #21037 
- jenkins tests pass
- see more details in the signoff of #21037 https://github.com/cms-sw/cmssw/pull/21037#issuecomment-346484906

**Comment by kpedro88** [comment]
+1

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/21045	1b7ef0f
- re-signing after a technical update since the last signoff by reco; changes are the same as in the master branch PR #21037 
- jenkins tests pass

**Comment by davidlange6** [comment]
merge

**Pull request merged**